app/src/main/java/com/brandon/forzenbook/view/composables/GeneralComposables.kt

```diff
@@ -157,48 +183,216 @@ fun InputInfoTextField(
+            modifier = modifier
+                .padding(
+                    dimens.paddingLarge,
+                    dimens.minimumTouchTargetPadding,
+                    dimens.paddingLarge,
+                    dimens.minimumTouchTargetPadding
+                )
+                .alpha(if (isEnabled) ENABLED_ALPHA else DISABLED_ALPHA)
+                .fillMaxWidth()
+                .imePadding(),
+            onValueChange = {
+                if (currentText.length == characterLimit) {
+                    onMaxCharacterLength(true)
+                } else {
+                    onTextChange(it)
+                    onMaxCharacterLength(false)
+                }
+            },
+            enabled = isEnabled,
+            textStyle = MaterialTheme.typography.labelMedium,
+            singleLine = true,
+            placeholder = {
+                HintText(text = hint)
+            },
+            visualTransformation = visualTransformation,
+            keyboardOptions = keyboardOptions,
+            colors = TextFieldDefaults.textFieldColors(
+                textColor = MaterialTheme.colorScheme.onPrimaryContainer,
+                backgroundColor = MaterialTheme.colorScheme.primaryContainer,
+            ),
+        )
+        Box(
+            modifier = Modifier
+                .matchParentSize()
+                .alpha(NO_ALPHA)
+                .clickable { onClickEvent() }
+        )
+    }
+}
+
 @Composable
 fun RedirectText(
     text: String,
+    isEnabled: Boolean = true,
     onClick: () -> Unit,
 ) {
     Text(
         modifier = Modifier
-            .padding(LocalDimens.current.paddingSmall)
-            .clickable { onClick() },
+            .clickable { if (isEnabled) onClick() }
+            .padding(LocalDimens.current.minimumTouchTargetPadding)
+            .alpha(if (isEnabled) ENABLED_ALPHA else DISABLED_ALPHA),
         text = text,
-        style = MaterialTheme.typography.bodyMedium,
+        style = MaterialTheme.typography.labelLarge,
         color = MaterialTheme.colorScheme.onSecondary,
     )
 }
 
 @Composable
-fun DimBackgroundLoading() {
-    Surface(
-        color = Color.Black.copy(alpha = LOADING_BACKGROUND_ALPHA),
-        modifier = Modifier
-            .fillMaxSize(),
-    ) {
+fun ImageTitle(
+    @DrawableRes imageId: Int,
+    text: String,
+) {
+    Image(
+        painter = painterResource(id = imageId),
+        contentDescription = LocalContext.current.getString(R.string.loginImageDescription),
+        modifier = Modifier.size(LocalDimens.current.largeIcon)
+    )
+    TitleText(
+        text = text
+    )
+}
+
+@Composable
+fun TitleText(
+    text: String,
+) {
+    Text(
+        modifier = Modifier.padding(LocalDimens.current.paddingSmall),
+        text = text,
+        style = MaterialTheme.typography.displayMedium,
+        overflow = TextOverflow.Ellipsis,
+        maxLines = TEXT_FIELD_MAX_LINES,
+    )
+}
+
+@Composable
+fun AlertDialog(
```

Comment:
I wrote about a launched effect earlier for an AlertDialog thinking it was the Android View alert dialog. You probably don't need a launchedeffect for this, but definitely do for android ones like the date dialog

Reply:
made the change to use a Launched Effect.